x/pricefeed/simulation/operations/msg.go

```diff
@@ -0,0 +1,157 @@
+package operations
+
+import (
+	"fmt"
+	"math/rand"
+	"strings"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/simulation"
+
+	"github.com/kava-labs/kava/x/pricefeed"
+	"github.com/kava-labs/kava/x/pricefeed/keeper"
+	"github.com/kava-labs/kava/x/pricefeed/types"
+)
+
+var (
+	noOpMsg     = simulation.NoOpMsg(pricefeed.ModuleName) // TODO QUESTION what is the purpose of noopmsg?
+	stableDenom = "usdx"                                   // TODO QUESTION is this necessary ?
+)
+
+// SimulateMsgUpdatePrices updates the prices of various assets
+// TODO - MUST include BNB, USDX, BTC as HARD REQUIREMENTS - asset prices needed by CDP sims
+func SimulateMsgUpdatePrices(keeper keeper.Keeper) simulation.Operation {
+	// get a pricefeed handler
+	handler := pricefeed.NewHandler(keeper)
+
+	// There's two error paths
+	// - return a OpMessage, but nil error - this will log a message but keep running the simulation
+	// - return an error - this will stop the simulation (I think)
+	return func(r *rand.Rand, app *baseapp.BaseApp, ctx sdk.Context, accs []simulation.Account) (
+		simulation.OperationMsg, []simulation.FutureOperation, error) {
+
+		// OVERALL LOGIC IDEA:
+		// (1) RANDOMLY PICK AN ASSET OUT OF BNB AN BTC [TODO QUESTION - PRESUMABLY USDX IS EXCLUDED AS IT IS A STABLE DENOM?]
+		// (2) GET THE CURRENT PRICE OF THAT ASSET IN USD
+		// (3) GENERATE A RANDOM NUMBER IN THE RANGE 0.8-1.2 (UNIFORM DISTRIBUTION) [TODO POTENTIAL UPDATE MAKE IT NORMALLY DISTRIBUTED ACROSS A WIDER RANGE?]
+		// (4) MULTIPLY THE CURRENT PRICE BY THE RANDOM NUMBER
+		// (5) POST THE NEW PRICE TO THE KEEPER
+
+		// pick a random asset out of BNB and BTC
+		assetCode := pickRandomAsset(ctx, keeper, r)
+
+		fmt.Printf("Picked asset: %s\n", assetCode)
+
+		//GET THE CURRENT PRICE OF THE ASSET
+		marketID := strings.ToLower(assetCode + ":usd") // convert to lower case
```

Comment:
return the market from `pickRandomAsset` use `market.MarketID`

Reply:
🆗